Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -376,6 +369,55 @@ CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* compo
 
 #pragma region WIC_HELPERS
 
+IWICBitmap* _CGImageGetImageSource(CGImageRef image) {
+    RETURN_NULL_IF(!image);
+    return image->ImageSource().Get();
+}
+
+DisplayTexture* _CGImageGetDisplayTexture(CGImageRef image) {
+    RETURN_NULL_IF(!image);
+    CGIWICBitmap* iwicImg = dynamic_cast<CGIWICBitmap*>(image->ImageSource().Get());
+    RETURN_NULL_IF(!iwicImg);
+    return iwicImg->GetDisplayTexture()->GetTexture();
+}
+
+// Return the data pointer to the Image data.
+void* _CGImageGetRawBytes(CGImageRef image) {
+    RETURN_NULL_IF(!image);
+    return image->Data();
+}
+
+CGImageRef _CGImageCreateWithWICBitmap(IWICBitmap* bitmap) {
+    RETURN_NULL_IF(!bitmap);
+    CGImageRef imageRef = __CGImage::CreateInstance();
+    imageRef->SetImageSource(bitmap);
+
+    return imageRef;
+}
+
+CGImageRef _CGImageCreateWithPixelFormat(CGImageRef image, WICPixelFormatGUID pixelFormat) {
+    RETURN_NULL_IF(!image);
+    if (IsEqualGUID(image->PixelFormat(), pixelFormat)) {
+        CGImageRetain(image);
+        return image;
+    }
+
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+    ComPtr<IWICFormatConverter> converter;
+    RETURN_NULL_IF_FAILED(imageFactory->CreateFormatConverter(&converter));
+
+    RETURN_NULL_IF_FAILED(converter->Initialize(
+        image->ImageSource().Get(), pixelFormat, WICBitmapDitherTypeNone, nullptr, 0.f, WICBitmapPaletteTypeMedianCut));
+
+    ComPtr<IWICBitmap> convertedImage;
+    RETURN_NULL_IF_FAILED(imageFactory->CreateBitmapFromSource(converter.Get(), WICBitmapCacheOnDemand, &convertedImage));
+
+    CGImageRef imageRef = __CGImage::CreateInstance();
+    imageRef->SetImageSource(convertedImage);
```

Comment:
uh I'd rather not increase it by one more call.
This should be fixed when we go constructor based (as well as other places).

---
In reply to: [87101870](https://github.com/Microsoft/WinObjC/pull/1332#discussion_r87101870) [](ancestors = 87101870)